"error: unmappable character for encoding windows-1252" when running tests

**Describe the bug**
When running the master branch tests at https://github.com/MobilityData/gtfs-validator/commit/455dbdbf39999f610ecdc13ea8807d10044252a8 on Windows I get:

```
> Task :core:compileTestJava
C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\CsvFileTest.java:153: error: unmappable character (0x8F) for encoding windows-1252
    InputStream inputStream = toInputStream("stop_id,stop_name\n" + "s1,Перва�? о�?тановка\n");
                                                                                   ^
C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\CsvFileTest.java:153: error: unmappable character (0x81) for encoding windows-1252
    InputStream inputStream = toInputStream("stop_id,stop_name\n" + "s1,Перва�? о�?тановка\n");
                                                                                        ^
C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\CsvFileTest.java:155: error: unmappable character (0x8F) for encoding windows-1252
    assertThat(csvFile.iterator().next().asString(1)).isEqualTo("Перва�? о�?тановка");
                                                                            ^
C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\CsvFileTest.java:155: error: unmappable character (0x81) for encoding windows-1252
    assertThat(csvFile.iterator().next().asString(1)).isEqualTo("Перва�? о�?тановка");
                                                                                 ^
C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\RowParserTest.java:196: error: unmappable character (0x9D) for encoding windows-1252
    RowParser parser = createParser("קו�?");
                                          ^
C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\RowParserTest.java:197: error: unmappable character (0x9D) for encoding windows-1252
    // .קו�? :the .COM equivalent in Hebrew
```

**How we reproduce the bug**
Steps to reproduce the behaviour:
1. Run the unit tests on Windows 10

**Expected behaviour**
They should run without errors

**Observed behaviour**
I get the above error

**Environment versions**
- validator version: master branch tests at https://github.com/MobilityData/gtfs-validator/commit/455dbdbf39999f610ecdc13ea8807d10044252a8
- Java version: JDK 11
- OS versions: Windows 10

```
IntelliJ IDEA 2020.2.4 (Ultimate Edition)
Build #IU-202.8194.7, built on November 24, 2020
Runtime version: 11.0.9+11-b944.49 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1955M
Cores: 8
Non-Bundled Plugins: Lombook Plugin, idea.plugin.protoeditor, io.protostuff.protostuff-jetbrains-plugin, net.seesharpsoft.intellij.plugins.csv, org.jetbrains.kotlin
```

Do we really need to take into account legacy systems that don't support
Unicode? This is not inclusive. There are major world languages that don't
use Latin script (Chinese, Hindi, Russian) or need diacritics (French,
Spanish, Portuguese), so refuse to support them would be at least impolite.

Den tis 16 feb. 2021 08:53Sean Barbeau <notifications@github.com> skrev:

> *Describe the bug*
> When running the master branch tests at 455dbdb
> <https://github.com/MobilityData/gtfs-validator/commit/455dbdbf39999f610ecdc13ea8807d10044252a8>
> on Windows I get:
>
> > Task :core:compileTestJava
>
> C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\CsvFileTest.java:153: error: unmappable character (0x8F) for encoding windows-1252
>
>     InputStream inputStream = toInputStream("stop_id,stop_name\n" + "s1,Перва�? о�?тановка\n");
>
>                                                                                    ^
>
> C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\CsvFileTest.java:153: error: unmappable character (0x81) for encoding windows-1252
>
>     InputStream inputStream = toInputStream("stop_id,stop_name\n" + "s1,Перва�? о�?тановка\n");
>
>                                                                                         ^
>
> C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\CsvFileTest.java:155: error: unmappable character (0x8F) for encoding windows-1252
>
>     assertThat(csvFile.iterator().next().asString(1)).isEqualTo("Перва�? о�?тановка");
>
>                                                                             ^
>
> C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\CsvFileTest.java:155: error: unmappable character (0x81) for encoding windows-1252
>
>     assertThat(csvFile.iterator().next().asString(1)).isEqualTo("Перва�? о�?тановка");
>
>                                                                                  ^
>
> C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\RowParserTest.java:196: error: unmappable character (0x9D) for encoding windows-1252
>
>     RowParser parser = createParser("קו�?");
>
>                                           ^
>
> C:\git-projects\gtfs-validator\core\src\test\java\org\mobilitydata\gtfsvalidator\parsing\RowParserTest.java:197: error: unmappable character (0x9D) for encoding windows-1252
>
>     // .קו�? :the .COM equivalent in Hebrew
>
>
> *How we reproduce the bug*
> Steps to reproduce the behaviour:
>
>    1. Run the unit tests on Windows 10
>
> *Expected behaviour*
> They should run without errors
>
> *Observed behaviour*
> I get the above error
>
> *Environment versions*
>
>    - validator version: master branch tests at 455dbdb
>    <https://github.com/MobilityData/gtfs-validator/commit/455dbdbf39999f610ecdc13ea8807d10044252a8>
>    - Java version: JDK 11
>    - OS versions: Windows 10
>
> IntelliJ IDEA 2020.2.4 (Ultimate Edition)
>
> Build #IU-202.8194.7, built on November 24, 2020
>
> Runtime version: 11.0.9+11-b944.49 amd64
>
> VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
>
> Windows 10 10.0
>
> GC: ParNew, ConcurrentMarkSweep
>
> Memory: 1955M
>
> Cores: 8
>
> Non-Bundled Plugins: Lombook Plugin, idea.plugin.protoeditor, io.protostuff.protostuff-jetbrains-plugin, net.seesharpsoft.intellij.plugins.csv, org.jetbrains.kotlin
>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MobilityData/gtfs-validator/issues/749>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAG6IJH52ZCB344SRVMMOWDS7GJVJANCNFSM4XVLUS3Q>
> .
>

--- a/core/build.gradle
+++ b/core/build.gradle
@@ -22,6 +22,8 @@ group 'org.mobilitydata'
 version "gtfs-validator-${System.getenv("versionTag")}"
 
 sourceCompatibility = JavaVersion.VERSION_11
+compileJava.options.encoding = "UTF-8"
+compileTestJava.options.encoding = "UTF-8"
 
 repositories {
     mavenCentral()

--- a/core/build.gradle
+++ b/core/build.gradle
@@ -22,7 +22,6 @@ group 'org.mobilitydata'
 version "gtfs-validator-${System.getenv("versionTag")}"
 
 sourceCompatibility = JavaVersion.VERSION_11
-compileJava.options.encoding = "UTF-8"
 compileTestJava.options.encoding = "UTF-8"
 
 repositories {

